src/vision/frc_vision_final.py

```diff
@@ -95,82 +73,56 @@ def newLine(
-    frames = rs2.composite_frame(pipe.wait_for_frames())
-    frame = rs2.video_frame(frames.get_color_frame())
-    depth = rs2.depth_frame(frames.get_depth_frame())
-
-    if not frame:
-        continue
+    if not debugging:
+        # Waits to get frames, and gets information about the frame.
+        frames = rs2.composite_frame(pipe.wait_for_frames())
+        frame = rs2.video_frame(frames.get_color_frame())
+        depth = rs2.depth_frame(frames.get_depth_frame())
 
-    IMG = np.asanyarray(frame.get_data())
+        if not frame:
+            continue
 
-    isShooting = sd.getBoolean("Is Shooting", defaultValue=False)
-    if not isShooting == OldStream:
-        swapStream(isShooting)
-
-    if isShooting:
-        # Convert from RGB to HSV, helps with filtering
-        HSV = cv2.cvtColor(IMG, cv2.COLOR_BGR2HSV)
+        IMG = np.asanyarray(frame.get_data())
 
-        # Define upper and lower bounds for HSV variables
-        LOWER_COLOR = np.array([70, 80, 255])
-        UPPER_COLOR = np.array([95, 180, 255])
-
-        # Create mask within hsv range
-        MASK = cv2.inRange(HSV, LOWER_COLOR, UPPER_COLOR)
-
-        # Blur image (buffer for pixel imperfections)
-        MEDIAN = cv2.medianBlur(MASK, 5)
-
-        # Edge detection for use in line detection
-        MED_EDGES = cv2.Canny(MEDIAN, 50, 150)
-
-        if debugging:
-            BLUR = cv2.GaussianBlur(MASK, (3, 3), 0)
-
-            BLUR_EDGES = cv2.Canny(BLUR, 100, 200)
-            MASK_EDGES = cv2.Canny(MASK, 100, 200)
-
-            # Empty image for drawing lines (testing)
-            FILTERED_LINE_IMG = np.zeros((HEIGHT, WIDTH, 3), np.uint8)
-            LINE_IMG = np.zeros((HEIGHT, WIDTH, 3), np.uint8)
+        isShooting = sd.getBoolean("Is Shooting", defaultValue=False)
+        if isShooting != OldStream:
+            swapStream(isShooting, s)
+    else:
+        IMG = cv2.imread('Elbit3_Color.png')
+        isShooting = True
 
-        # Find lines in selected image
-        LINES = cv2.HoughLinesP(MED_EDGES, 1, radians(0.5), 25, maxLineGap=25)
+    lines = imgfilter(IMG, LOWER_COLOR, UPPER_COLOR, debugging, WIDTH, HEIGHT)
 
+    if isShooting:
         # If there are lines:
-        if LINES is not None:
-            NUM_LINES = len(LINES)
+        if lines is not None:
+            NUM_LINES = len(lines)
             FILTERED_LINES = []
             X_TOTAL = 0
             Y_TOTAL = 0
 
-            for NEW_LINE in LINES:
+            for NEW_LINE in lines:
                 x1, y1, x2, y2 = NEW_LINE[0]
                 new_slope = degrees(np.arctan((y2 - y1) / (x2 - x1)))
 
                 # Checks if we have verified lines, and makes a new line based on that.
                 if FILTERED_LINES:
                     if (
-                        new_slope < -ANGLE_THRESHOLD or new_slope > ANGLE_THRESHOLD
+                        ANGLE_HIGH_THRESHOLD > abs(new_slope) > ANGLE_LOW_THRESHOLD
```

Comment:
Instead of calling a function within that block, set a variable. Then do `if variable: newLine()`.

Then remove the `newLine` function and just replace it with the contents, since it's more problematic than useful. It takes nine parameters...